### ISSUE TITLE
D102: RUF100 ping-pong

### ISSUE THREAD

**Issue by smurfix** [astral-sh/ruff#20639]
### Summary

This complains with RUF100. When I remove the comment, it instead complains with D102.

```
"Test."
from __future__ import annotations


class Foo:
    "Test."

    def test(self):  # noqa:D102
        return 1
```

**Comment by ntBre** [comment]
I can't reproduce this one in the [playground](https://play.ruff.rs/b2916cf8-8a16-4141-bf34-31aa4a0df39d) with both D102 and RUF100 selected. Are you possibly selecting only RUF100? It takes into account the other selected rules to know if a comment is unused:

```
> ruff check --select RUF100 --output-format=concise
try.py:8:22: RUF100 [*] Unused `noqa` directive (non-enabled: `D102`)
Found 1 error.
[*] 1 fixable with the `--fix` option.
> ruff check --select RUF100,D102
All checks passed!
```

**Comment by m-u-xyz** [comment]
Ugh. No, but related. My fault. Sorry about the noise.
